### PR TITLE
added methods with context + upgrad go version

### PR DESCRIPTION
**Pull request by crayonwow**
add methods that allows pass context to calls
upgraded go version up to 1.19

### CODE REVIEW

File: bot.go [hunk 1]
```diff
@@ -460,12 +470,24 @@ func (bot *BotAPI) GetUpdatesChan(config UpdateConfig) UpdatesChannel {
 	return ch
 }
 
+// GetUpdatesChan starts and returns a channel for getting updates.
+func (bot *BotAPI) GetUpdatesChan(config UpdateConfig) UpdatesChannel {
+	ctx, cancel := context.WithCancel(context.Background())
+	bot.mu.Lock()
+	bot.stoppers = append(bot.stoppers, cancel)
```
Comment (crayonwow) [inline bot.go:477]: in case of multiple calls we made a stoppers as slice of context.CancelFunc and call them under the lock

File: bot_test.go [hunk 2]
```diff
@@ -1001,32 +1001,30 @@ func TestCommands(t *testing.T) {
 	}
 }
 
-// TODO: figure out why test is failing
-//
-// func TestEditMessageMedia(t *testing.T) {
-// 	bot, _ := getBot(t)
+func TestEditMessageMedia(t *testing.T) {
```
Comment (crayonwow) [inline bot_test.go:1004]: actually it works
idk why

File: configs.go [hunk 3]
```diff
@@ -2566,3 +2566,20 @@ func prepareInputMediaForFiles(inputMedia []interface{}) []RequestFile {
 
 	return files
 }
+
+type chattable struct {
```
Comment (crayonwow) [inline configs.go:2570]: simple configurable structure that implements Chattable interface

**Comment by OsoianMarcel** [comment]
The initiative to implement the Context is welcome, thank you. Without context, the long polling HTTP requests are not cancelable, so for graceful shutdown, the app needs to wait from 0 to 60 seconds.

Dear author, thank you for the library, I find it the most easy to use.

**Comment by npazosmendez** [comment]
hello @crayonwow! the project seemds abandoned, but there's a fork keeping it alive: https://github.com/OvyFlash/telegram-bot-api . would be open to sending these changes to that fork?

**Pull request closed without merging**